### PR TITLE
feat: convert accordion to radix primitive

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* refactors Accordion to use Radix Primitive
* adds easings to the theme to use with css animations
* adds a `ViewSource` component to the MDX list so we can easily show the source of our components without changing 1000s of lines

### Related issue(s)/PR(s)

* resolves DX-1206

### CODE REVIEW

File: docs/.storybook/main.ts [hunk 1]
```diff
@@ -21,8 +23,9 @@ const config: StorybookConfig = {
       },
     },
   },
-  viteFinal: (config) => {
-    if (config.mode !== 'production') {
+  viteFinal: (config, { configType }) => {
+    console.log(configType);
```
Comment (jhoward1994) [inline docs/.storybook/main.ts:27]: Remove this?

Comment (joshuaellis) [reply docs/.storybook/main.ts:27]: Yes! Sharp eyes

Comment (joshuaellis) [inline docs/.storybook/main.ts:27]:
```suggestion
```

File: docs/stories/components/Accordion.mdx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+```js
+import { Accordion } from '@strapi/design-system';
+```
+
+## Props
+
+### Accordion
+
+<ArgTypes of={Accordion.Root} />
```
Comment (jhoward1994) [inline docs/stories/components/Accordion.mdx:35]: I'm seeing this in the story here http://localhost:6006/?path=/docs/components-accordion--docs

Comment (joshuaellis) [reply docs/stories/components/Accordion.mdx:35]: yeah they dont work yet :( I need to make a custom component & learn how to do it... SOON

File: docs/stories/components/Accordion.mdx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+Optionally, an icon can be added to the left of the trigger title.
+
+<Canvas of={AccordionStories.WithIcon} />
+
+### With Actions
+
+When actions are provided, they will be displayed to the right of the trigger title, you should use `IconButton`
+components for the actions ensuring they are accessible with a `label` prop. The iternals of the component will handle
+the visual adjustments we expect from actions.
```
Comment (jhoward1994) [inline docs/stories/components/Accordion.mdx:104]:
```suggestion
components for the actions ensuring they are accessible with a `label` prop. The internals of the component will handle
the visual adjustments we expect from actions.
```

File: packages/design-system/src/components/Accordion/Accordion.tsx [hunk 4]
```diff
@@ -1,148 +1,357 @@
...
+  ({ caretPosition = 'left', description, icon: Icon, children, ...restProps }, forwardedRef) => {
+    const { size } = useAccordion('Trigger');
 
-type AccordionWrapperProps = Pick<AccordionProps, 'expanded' | 'disabled' | 'variant' | 'error'>;
+    return (
+      <AccordionTrigger $caretPosition={caretPosition} $size={size} ref={forwardedRef} {...restProps}>
+        {caretPosition === 'left' ? (
+          <TriggerIcon $size={size}>
+            <CaretDown width={size === 'S' ? '1.2rem' : '1.6rem'} height={size === 'S' ? '1.2rem' : '1.6rem'} />
+          </TriggerIcon>
```
Comment (jhoward1994) [inline packages/design-system/src/components/Accordion/Accordion.tsx:126]: Is it worth having this as a const rather than repeating it for left and right?

Comment (joshuaellis) [reply packages/design-system/src/components/Accordion/Accordion.tsx:126]: we could, it's a much of a muchness... I don't think we'll be messing with this at all and it's pretty simple

**Review by jhoward1994 — APPROVED** [review]
ted-lasso-im-flattered-gif-by-apple-tv

**Pull request merged**